sklearn/inspection/partial_dependence.py

```diff
@@ -0,0 +1,664 @@
+        :term:`predict`. By default, :term:`predict_proba` is tried first
+        and we revert to :term:`decision_function` if it doesn't exist. If
+        ``method`` is 'recursion', the response is always the output of
+        :term:`decision_function`.
+    n_cols : int, optional (default=3)
+        The maximum number of columns in the grid plot.
+    grid_resolution : int, optional (default=100)
+        The number of equally spaced points on the axes of the plots, for each
+        target feature.
+    percentiles : tuple of float, optional (default=(0.05, 0.95))
+        The lower and upper percentile used to create the extreme values
+        for the PDP axes. Must be in [0, 1].
+    method : str, optional (default='auto')
+        The method to use to calculate the partial dependence predictions:
+
+        - 'recursion' is only supported for objects inheriting from
+          `BaseGradientBoosting`, but is more efficient in terms of speed.
+          With this method, ``X`` is optional and is only used to build the
+          grid and the partial dependences are computed using the training
+          data. This method does not account for the ``init`` predicor of
+          the boosting process, which may lead to incorrect values (see
+          warning below. With this method, the target response of a
+          classifier is always the decision function, not the predicted
+          probabilities.
+
+        - 'brute' is supported for any estimator, but is more
+          computationally intensive.
+
+        - If 'auto', then 'recursion' will be used for
+          ``BaseGradientBoosting`` estimators with ``init=None``, and
+          'brute' for all other.
+
+        Unlike the 'brute' method, 'recursion' does not account for the
+        ``init`` predictor of the boosting process. In practice this still
+        produces the same plots, up to a constant offset in the target
+        response.
+    n_jobs : int, optional (default=None)
+        The number of CPUs to use to compute the partial dependences.
+        ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        ``-1`` means using all processors. See :term:`Glossary <n_jobs>`
+        for more details.
+    verbose : int, optional (default=0)
+        Verbose output during PD computations.
+    fig : Matplotlib figure object, optional (default=None)
+        A figure object onto which the plots will be drawn, after the figure
+        has been cleared.
+    line_kw : dict, optional
+        Dict with keywords passed to the ``matplotlib.pyplot.plot`` call.
+        For one-way partial dependence plots.
+    contour_kw : dict, optional
+        Dict with keywords passed to the ``matplotlib.pyplot.plot`` call.
+        For two-way partial dependence plots.
+    **fig_kw : dict, optional
+        Dict with keywords passed to the figure() call.
+        Note that all keywords not recognized above will be automatically
```

Comment:
Do we need this? If the user wants custom figure features, they can pass `fig`. This is the typical unsafe use of kwargs that matplotlib uses where you can't tell where things will be routed. I don't like it. Gaël has certainly been loudly against it.

Reply:
This is just for "backward compatibility".

I'm happy to see it go.

Reply:
If we're able to be passed a fig, then we should remove it.